lib/src/widgets/search.dart

```diff
@@ -32,73 +32,47 @@ class ArticleSearch extends SearchDelegate<Article> {
 
   @override
   Widget buildResults(BuildContext context) {
-    return StreamBuilder<UnmodifiableListView<Article>>(
-      stream: articles,
-      builder:
-          (context, AsyncSnapshot<UnmodifiableListView<Article>> snapshot) {
-        if (!snapshot.hasData) {
-          return Center(
-              child: Text(
-            'No data!',
-          ));
-        }
+    var results = articles
+        .where((a) => a.title.toLowerCase().contains(query.toLowerCase()));
 
-        var results = snapshot.data
-            .where((a) => a.title.toLowerCase().contains(query.toLowerCase()));
-
-        return ListView(
-          children: results
-              .map<ListTile>((a) => ListTile(
-                    title: Text(a.title,
-                        style: Theme.of(context)
-                            .textTheme
-                            .subhead
-                            .copyWith(fontSize: 16.0)),
-                    leading: Icon(Icons.book),
-                    onTap: () async {
-                      if (await canLaunch(a.url)) {
-                        await launch(a.url);
-                      }
-                      close(context, a);
-                    },
-                  ))
-              .toList(),
-        );
-      },
+    return ListView(
+      children: results
+          .map<ListTile>((a) => ListTile(
+                title: Text(a.title,
+                    style: Theme.of(context)
+                        .textTheme
+                        .subhead
+                        .copyWith(fontSize: 16.0)),
+                leading: Icon(Icons.book),
+                onTap: () async {
+                  if (await canLaunch(a.url)) {
+                    await launch(a.url);
+                  }
+                  close(context, a);
+                },
+              ))
+          .toList(),
     );
   }
 
   @override
   Widget buildSuggestions(BuildContext context) {
-    return StreamBuilder<UnmodifiableListView<Article>>(
-      stream: articles,
-      builder:
-          (context, AsyncSnapshot<UnmodifiableListView<Article>> snapshot) {
-        if (!snapshot.hasData) {
-          return Center(
-              child: Text(
-            'No data!',
-          ));
-        }
+    final results = articles
+        .where((a) => a.title.toLowerCase().contains(query.toLowerCase()));
```

Comment:
might talk about how the removal of bloc removes these StreamBuilder layers because ChangeNotifier now tells us when we need to redraw